### ISSUE TITLE
Import of MusicXML files: Chord symbol didn't load correctly

### ISSUE THREAD

**Issue by WindyPigeon** [musescore/MuseScore#16638]
### Issue type

Engraving bug

### Bug description

When importing MusicXML files, MuseScore 4 removes the root name from the chord symbols.
You can test it yourself with test/harmony2.xml or other MusicXML files, and you will get similar results as the screenshot.

The problem probably comes from the following code block
https://github.com/musescore/MuseScore/blob/v4.0.1/src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp#L5215-L5239
MuseScore 4 has a logic error in handling of MusicXML's `root-step` tag.
According [MusicXML documentation on the `root-step` element](https://www.w3.org/2021/06/musicxml40/musicxml-reference/elements/root-step/), the `text` attribute in `root-step` element is not necessary, and if `text` attribute doesn't exist in `root-step` element it will use the element contents as the root name.

Below is my suggested fix.
Replace [line 5215 - line 5220]( https://github.com/musescore/MuseScore/blob/v4.0.1/src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp#L5215-L5220) of **importmxmlpass2.cpp** with following the code:
```c++
if ((_e.attributes().hasAttribute("text")) && (_e.attributes().value("text").toString() != "")) {
    step = _e.attributes().value("text").toString();
} else if (_e.readElementText != "") {
    step = _e.readElementText();
} else {
    invalidRoot = true;
}
```

I don't have a C++ development environment on my PC, please somebody help me to test the code and commit it to GitHub.

### Steps to reproduce

Import any MusicXML file that contains annotations for chord symbols.

### Screenshots/Screen recordings

![image](https://user-images.githubusercontent.com/62902049/222669408-22ad5603-a7ef-4005-84e5-446362e72814.png)

### MuseScore Version

4.0.1

### Regression

Yes, this used to work in Musescore 3.x and now is broken

### Operating system

Windows

**Comment by Jojo-Schmitz** [comment]
Isn't this a duplicate of  https://github.com/musescore/MuseScore/issues/13214 ?

**Comment by WindyPigeon** [comment]
> Isn't this a duplicate of  https://github.com/musescore/MuseScore/issues/13214 ?

Yes. I think it will be fixed soon, because the cause of the bug has been found.

**Comment by Jojo-Schmitz** [comment]
Hmm, not sure your analysis is correct though, that code is in since at lest Nov 17, 2020, 133fc82d0b3, so in 3.6 already, but that doesn't have this issue as far as I can tell

**Comment by Jojo-Schmitz** [comment]
But let's summon @lvinken on this ;-)

**Comment by MarcSabatella** [comment]
The cause *is* known; it's what I stated in that other issue.  Not the red herring that was *also* posted in that other issue, although it certainly looked like a good guess.

**Comment by WindyPigeon** [comment]
> The cause *is* known; it's what I stated in that other issue.  Not the red herring that was *also* posted in that other issue, although it certainly looked like a good guess.

Yes. You're right. Although the logic of that code block is weird for me, but it doesn't cause the broken.  Maybe other changing between 3.6 and 4.0 cause this problem, I'm now comparing the source code between 3.6 and 4.0 to find the culprit.

**Comment by MarcSabatella** [comment]
Again, it's exactly as I stated multiple times already - the style initialization code no longer loads the chord description file, so it needs to be done separately.  The only question is where that should be best done.  I think it makes sense to do it somewhere fairly central so each importer doesn't need to do it manually.  I just don't know where the right place would be.  See in particular my comment where I point directly to the relevant code - https://github.com/musescore/MuseScore/issues/13214#issuecomment-1379458168

**Comment by lvinken** [comment]
> Hmm, not sure your analysis is correct though, that code is in since at lest Nov 17, 2020, [133fc82](https://github.com/musescore/MuseScore/commit/133fc82d0b36c5e025f8bef077c342cf7f32d94a), so in 3.6 already, but that doesn't have this issue as far as I can tell.
> 
> But let's summon @lvinken on this ;-)

Actually I agree with Marc's analysis, so I have nothing to add here.

**Comment by ExcPoint** [comment]
I am running into the same issue in MuseScore 4.0. MusicXML is not rendering correctly when opened in all formats, Compressed (.mxl), Uncompressed (.musicxml & .xml).

**Comment by MarcSabatella** [comment]
Indeed, the issue is not fixed, but the workarounds explained previously continue to work (change appearance to jazz & back, or save & reload).

**Comment by asattely** [comment]
This issue is fixed by https://github.com/musescore/MuseScore/pull/17427

**Comment by Jojo-Schmitz** [comment]
Should get resolved by #18737 as far as I can tell
